plugins/migration/bin/fix-imports.js

```diff
@@ -0,0 +1,237 @@
+    newRequire: `const {dartGenerator} = require('blockly/dart');`,
+  },
+  {
+    import: 'blockly/javascript',
+    oldIdentifier: 'Blockly.JavaScript',
+    newIdentifier: 'javascriptGenerator',
+    newImport: `import {javascriptGenerator} from 'blockly/javascript';`,
+    newRequire: `const {javascriptGenerator} = require('blockly/javascript');`,
+  },
+  {
+    import: 'blockly/lua',
+    newIdentifier: 'luaGenerator',
+    newImport: `import {luaGenerator} from 'blockly/lua';`,
+    newRequire: `const {luaGenerator} = require('blockly/lua');`,
+  },
+  {
+    import: 'blockly/php',
+    newIdentifier: 'phpGenerator',
+    newImport: `import {phpGenerator} from 'blockly/php';`,
+    newRequire: `const {phpGenerator} = require('blockly/php');`,
+  },
+  {
+    import: 'blockly/python',
+    newIdentifier: 'pythonGenerator',
+    newImport: `import {pythonGenerator} from 'blockly/python';`,
+    newRequire: `const {pythonGenerator} = require('blockly/python');`,
+  },
+  {
+    import: 'blockly/blocks',
+    oldIdentifier: 'Blockly.libraryBlocks',
+    newIdentifier: 'libraryBlocks',
+    newImport:  `import {libraryBlocks} from 'blockly/blocks';`,
+    newRequire: `const {libraryBlocks} = require('blockly/blocks');`,
+  }
+]
+
+/**
+ * Migrates the contents of a particular file, renaming references and adding
+ * imports.
+ * 
+ * @param {string} contents The string contents of the file to migrate.
+ * @return {string} The migrated contents of the file.
+ */
+function createNewContents(contents) {
+  let newContents = contents;
+  for (const migrationData of database) {
+    newContents = fixImport(newContents, migrationData);
+  }
+  return newContents;
+}
+
+/**
+ * Migrates a particular import in a particular file. Renames references to
+ * where the import used to exist on the namespace tree, and adds a new import.
+ * 
+ * @param {string} contents The string contents of the file to migrate.
+ * @param {MigrationData} migrationData Data defining what to migrate and how.
+ * @return {string} The migrated contents of the file.
+ */
+function fixImport(contents, migrationData) {
+  const identifier = getIdentifier(contents, migrationData);
+  if (!identifier) return contents;
+  const newContents = replaceReferences(contents, migrationData, identifier);
+  if (newContents !== contents) return addImport(newContents, migrationData);
+  return contents;
+}
+
+function usesImportStatements(contents) {
+  return !!contents.match(/import.+'.+';/);
+}
+
+function usesRequireStatements(contents) {
+  return !!contents.match(/require\('.+'\);/)
+}
+
+function getIdentifier(contents, migrationData) {
+  if (usesImportStatements(contents)) {
+    const identifierMatch = contents.match(
+        new RegExp(`\\s(\\S*) from '${migrationData.import}'`));
+    if (!identifierMatch) return migrationData.oldIdentifier ?? null;
+    return identifierMatch[1];
+  }
+  if (usesRequireStatements(contents)) {
```

Comment:
I was going to say that even though it is not necessary and does spoil the aesthetics, I think it is helpful for readability to use an `else` here:

```suggestion
    return identifierMatch[1];
  } else if (usesRequireStatements(contents)) {
```
because otherwise one is left to puzzle out the consequences for files that use both`import` and `require`.

But then upon further consideration I think that there may actually be a bug here, because AFAIK it is quite legitimate for a node module to use both `import` and `require`, and it looks like this code might do the wrong thing in that case.

(Here and below at `addImport`.)